src/database/database.cpp

```diff
@@ -1,6 +1,149 @@
-#include <cstdint>
-#include <iostream>
-#include <mongocxx/instance.hpp>
-#include <mongocxx/stdx.hpp>
-#include <mongocxx/uri.hpp>
-#include <vector>
+#include "database.hpp"
+
+Database::Database()
+ : uri(mongocxx::uri("mongodb+srv://bakcyl324:Bakcyl768324@paragondatabase.jedczob.mongodb.net/")),
+ client(mongocxx::client(uri)),
+ database(client["ParagonApp"]) {}
+
+Database::~Database() {}
+
+//Checks if is there user with given Email
+bool Database::isThereUserWithThisEmail (const std::string& Email)
+{
+    auto collection = database["users"];
+
+    //searching for user with this email and returning result
+    auto cursor = collection.find_one(make_document(kvp("Email", Email)));
+    if(cursor) return true;
+    else return false;
+}
+
+//Checks if is there user with given Username
+bool Database::isThereUserWithThisUsername (const std::string& Username)
+{
+    auto collection = database["users"];
+
+    //searching for user with this username and returning result
+    auto cursor = collection.find_one(make_document(kvp("UserName", Username)));
+    if(cursor) return true;
+    else return false;
+}
+
+//creating new user
+//returns true if user was created
+//returns false if there is already user with that username or email(new user was not created)
+bool Database::createUser(const std::string& Username, const std::string& Password, const std::string& Email) 
+{
+    auto collection = database["users"];
+
+    
+    if(isThereUserWithThisUsername(Username) || isThereUserWithThisEmail(Email))
+    {
+        return false;
+    }//if there is already user with this username or email new user is not created
+    else
+    {
+        //Creates user with given elements
+        auto resultOfInsert = collection.insert_one(make_document(
+        kvp("UserName", Username),
+        kvp("Password", Password),
+        kvp("Email", Email)
+        ));
+    }
+    return true;
+}
+
+//deletes user
+//id can be either email or username
+//needs password of this user
+//returns true if user was deleted and false if user was not deleted(password is incorrect or there is no user with this id)
+bool Database::deleteUser(const std::string& id, const std::string& Password)
+{
+    auto collection = database["users"];
+
+    //Checking if the 'id' is a Username or Email
+    if(isThereUserWithThisUsername(id))//id is an Username
+    {
+        //Finding user by Username 
+        auto cursor = collection.find(make_document(kvp("UserName", id)));
+        for(auto&& doc : cursor)
```

Comment:
We have two very similar bunches of code, the first one [68:81] and the second one [87:100], please extract it to new function
The less code repetitions the better